apps/src/javalab/JavalabEditor.jsx

```diff
@@ -1,7 +1,7 @@
 import React from 'react';
```

Comment:
This file is getting pretty long. If anyone has thoughts on a good way to refactor it, let me know! Might be worth doing that as a follow-up to this work.

Reply:
One possibility is to have a `FileManager` and a `Editor` - the file manager would handle interactions with `projects`, and create/delete/rename. It would manage details about the files in memory. The Editor would handle the display logic. i.e. dark/light mode, dropdowns, and dialogs.

Reply:
that's a good idea. I think it would be worth splitting that out into a separate PR after this one to not bloat this one further.